contracts/votes/src/lib.rs

```diff
@@ -139,7 +139,10 @@ impl VotesTrait for VotesContract {
         Configuration::get(&env, dao_id)
     }
 
-    fn remove_configuration(env: Env, dao_id: Bytes) {
+    fn remove_configuration(env: Env, dao_id: Bytes, dao_owner: Address) {
+        let core_id = Self::get_core_id(env.clone());
+
+        verify_dao_owner(&env, &dao_id, dao_owner, core_id);
```

Comment:
Here you call back into the core contract, but you call this function also from destroy_dao, so that results in a re-entrancy error.